connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/executable/InboundExecutableRegistryImpl.java

```diff
@@ -92,32 +80,30 @@ public void publishEvent(InboundExecutableEvent event) {
     LOG.debug("Event added to the queue: {}", event);
   }
 
-  private final BiConsumer<Throwable, UUID> cancellationCallback =
-      (throwable, id) -> {
-        LOG.warn("Inbound connector executable has requested its cancellation", throwable);
-        var toCancel = executables.get(id);
-        if (toCancel == null) {
-          LOG.error("Inbound connector executable not found for the given ID: {}", id);
-          return;
-        }
-        if (toCancel instanceof Activated activated) {
-          activated.context().reportHealth(Health.down(throwable));
-          try {
-            activated.executable().deactivate();
-          } catch (Exception e) {
-            LOG.error("Failed to deactivate connector", e);
-          }
-        } else {
-          LOG.error(
-              "Attempted to cancel an inbound connector executable that is not in the active state: {}",
-              id);
-        }
-      };
-
   void handleEvent(InboundExecutableEvent event) {
     switch (event) {
       case InboundExecutableEvent.Activated activated -> handleActivated(activated);
       case Deactivated deactivated -> handleDeactivated(deactivated);
+      case InboundExecutableEvent.Cancelled cancelled -> handleCancelled(cancelled);
+    }
+  }
+
+  private void handleCancelled(InboundExecutableEvent.Cancelled cancelled) {
+    RegisteredExecutable executable = this.executables.get(cancelled.uuid());
+    executable = this.batchExecutableProcessor.cancelExecutable(executable, cancelled.throwable());
+    this.executables.replace(cancelled.uuid(), executable);
+    if (cancelled.throwable() instanceof ConnectorRetryException retryException) {
+      this.batchExecutableProcessor
+          .restartFromContext(executable, retryException)
+          .thenAccept(
+              registeredExecutable ->
+                  this.executables.replace(cancelled.uuid(), registeredExecutable))
+          .exceptionally(
+              throwable -> {
+                LOG.error(
+                    "The inbound connector could not be restarted, {}", throwable.getMessage());
```

Comment:
I would log the exception here, to avoid the issue we have with Kafka and other connectors (exception message hidden)

Reply:
Done